plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/expression/OperatorExpEmbeddingBuilder.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2010-2024 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.embeddings.expression
+
+import org.jetbrains.kotlin.formver.domains.FunctionBuilder
+import org.jetbrains.kotlin.formver.domains.InjectionImageFunction
+import org.jetbrains.kotlin.formver.embeddings.types.*
+import org.jetbrains.kotlin.formver.viper.ast.*
+
+
+class OperatorExpEmbeddingBuilder {
+    private var runtimeTypeFunctionName: String? = null
+    private var viperApplicable: Applicable? = null
+    private var callableType: FunctionTypeEmbedding? = null
+    private var additionalConditions: (FunctionBuilder.() -> Unit)? = null
+
+    fun complete(): OperatorExpEmbeddingTemplate? {
+        val callableType = callableType ?: error("Signature not specified to buildExpEmbedding.")
+        val argumentTypes = callableType.formalArgTypes
+        val returnType = callableType.returnType
+        val refsOperation = InjectionImageFunction(
```

Comment:
Why is the line with the arguments size check is removed? Are there any non unary or binary ops?